File: frameNet/src/main/java/org/sqlunet/framenet/sql/FnLayerQuery.java
package org.sqlunet.framenet.sql;

import android.database.sqlite.SQLiteDatabase;
import android.support.annotation.Nullable;

import org.sqlunet.framenet.Utils;
import org.sqlunet.sql.DBQuery;

import java.util.List;

/**
 * FrameNet layer query
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
class FnLayerQuery extends DBQuery
{
	/**
	 * Constructor
	 *
	 * @param connection connection
	 * @param targetId   target id
	 */
	@SuppressWarnings("boxing")
	FnLayerQuery(final SQLiteDatabase connection, final long targetId, final String query)
	{
		super(connection, query);
		setParams(targetId);
	}

	/**
	 * Get the layer id from the result set
	 *
	 * @return the layer id from the result set
	 */
	public long getLayerId()
	{
		assert this.cursor != null;
		return this.cursor.getLong(0);
	}

	/**
	 * Get the layer type from the result set
	 *
	 * @return the layer type from the result set
	 */
	public String getLayerType()
	{
		assert this.cursor != null;
		return this.cursor.getString(1);
	}

	/**
	 * Get the layer annoSet id from the result set
	 *
	 * @return the layer annoSet id from the result set
	 */
	public long getAnnoSetId()
	{
		assert this.cursor != null;
		return this.cursor.getLong(2);
	}

	/**
	 * Get the layer rank from the result set
	 *
	 * @return the layer rank from the result set
	 */
	public int getRank()
	{
		assert this.cursor != null;
		return this.cursor.getInt(3);
	}

	/**
	 * Get the labels from the result set
	 *
	 * @return the labels from the result set
	 */
	@Nullable
	public List<FnLabel> getLabels()
	{
		assert this.cursor != null;
		return Utils.parseLabels(this.cursor.getString(4));
	}
}

File: frameNet/src/main/java/org/sqlunet/framenet/sql/FnLayerQuery.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.framenet.sql

import android.database.sqlite.SQLiteDatabase
import org.sqlunet.framenet.Utils.parseLabels
import org.sqlunet.sql.DBQuery

/**
 * FrameNet layer query
 *
 * @param connection connection
 * @param targetId   target id
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
internal open class FnLayerQuery(connection: SQLiteDatabase, targetId: Long, query: String) : DBQuery(connection, query) {

    init {
        setParams(targetId)
    }

    /**
     * Layer id from the result set
     */
    val layerId: Long
        get() = cursor!!.getLong(0)

    /**
     * Layer type from the result set
     */
    val layerType: String
        get() = cursor!!.getString(1)

    /**
     * Layer annoSet id from the result set
     */
    val annoSetId: Long
        get() = cursor!!.getLong(2)

    /**
     * Layer rank from the result set
     */
    val rank: Int
        get() = cursor!!.getInt(3)

    /**
     * Labels from the result set
     */
    val labels: List<FnLabel>?
        get() = parseLabels(cursor!!.getString(4))
}
